assets/js/translator.js

```diff
@@ -558,6 +564,25 @@ function populateTranslationList() {
     }
 }
 
+function getSrcNumCols(numSrcCols) {
```

Comment:
I'm pretty confident you can eliminate the complex loops here and use some math instead :)